src/Sylius/Bundle/UserBundle/EventListener/UserDeleteListener.php

```diff
@@ -56,4 +55,18 @@ public function deleteUser(ResourceControllerEvent $event): void
             $flashBag->add('error', 'Cannot remove currently logged in user.');
         }
     }
+
+    private function isTryingToDeleteLoggedInAdminUser(UserInterface $user): bool
+    {
+        $isTryingToDeleteLoggedInUser = false;
+        if ($user instanceof AdminUserInterface) {
+            $token = $this->tokenStorage->getToken();
+            if ($token) {
+                $loggedUser = $token->getUser();
+                $isTryingToDeleteLoggedInUser = ($loggedUser instanceof AdminUserInterface) && ($loggedUser->getId() === $user->getId());
+            }
+        }
+
+        return $isTryingToDeleteLoggedInUser;
```

Comment:
I would say we need a little bit of refactoring with early returns here:

```php
$token = $this->tokenStorage->getToken();
if (!$user instanceof AdminUserInterface || $token === null) {
    return false;
}

if ($loggedUser instanceof AdminUserInterface && $loggedUser->getId() === $user->getId()) {
    return true;
}

return false;
```

wdyt?